v2/tools/generator/internal/codegen/pipeline/create_conversion_graph.go

```diff
@@ -50,6 +54,59 @@ func CreateConversionGraph(
 			return state.WithConversionGraph(graph), nil
 		})
 
+	stage.AddDiagnostic(
+		func(settings *DebugSettings, index int, state *State) error {
```

Comment:
Rather than defining this function inline, would it make sense to make it a standalone function and then just ref it here?

Could be defined in this file just a bit below?